packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -1,33 +1,55 @@
 <script lang="ts">
 import { Spinner } from '@podman-desktop/ui-svelte';
 
-type SearchFunction = (s: string) => Promise<string[]>;
-
-// the text displayed when no option is selected
-export let placeholder: string | undefined = undefined;
-export let required: boolean = false;
-export let delay: number = 200;
-export let disabled: boolean = false;
-export let initialFocus: boolean = false;
-export let id: string | undefined = undefined;
-export let name: string | undefined = undefined;
-export let error: boolean = false;
-
-export let searchFunction: SearchFunction = async (_s: string) => [];
-export let onChange = function (_s: string) {};
-export let onEnter = function () {};
+interface GroupItem {
+  values: string[];
+  group?: string;
+  sorted?: boolean;
+}
+
+interface Props {
+  placeholder?: string;
+  required?: boolean;
+  delay?: number;
+  disabled?: boolean;
+  initialFocus?: boolean;
+  id?: string;
+  name?: string;
+  error?: boolean;
+  className?: string;
+  onInputChange?: (s: string) => Promise<GroupItem[]>;
```

Comment:
It may feel easy to return from the onInputChange, but it is easier to split this in two.

```suggestion
  onInputChange?: (s: string) => void;
  values: GroupItem[]
```

So if we need to define initial value or change it based on external events, we can just change the values props without having to wait for an InputEvent.

The parent component should handle the logic of correlating the input and the values, this is a bit more work for it, as it need to have a `values = $state<GroupItem[]>(..)` but this is more svelte like